crates/oxc_language_server/src/commands.rs

```diff
@@ -0,0 +1,115 @@
+use log::error;
+use serde::Deserialize;
+use tower_lsp::{
+    jsonrpc::{self, Error},
+    lsp_types::{
+        request::ApplyWorkspaceEdit, ApplyWorkspaceEditParams, TextEdit, Url, WorkspaceEdit,
+    },
+};
+
+use crate::{capabilities::Capabilities, Backend};
+
+pub const LSP_COMMANDS: [WorkspaceCommands; 1] = [WorkspaceCommands::FixAll(FixAllCommand)];
+
+pub trait WorkspaceCommand {
+    fn command_id(&self) -> String;
+    fn available(&self, cap: Capabilities) -> bool;
+    type CommandArgs<'a>: serde::Deserialize<'a>;
+    async fn execute<'a>(
+        &self,
+        backend: &Backend,
+        args: Self::CommandArgs<'a>,
+    ) -> jsonrpc::Result<Option<serde_json::Value>>;
+}
+
+pub enum WorkspaceCommands {
+    FixAll(FixAllCommand),
+}
+
+impl WorkspaceCommands {
+    pub fn command_id(&self) -> String {
+        match self {
+            WorkspaceCommands::FixAll(c) => c.command_id().into(),
+        }
+    }
+    pub fn available(&self, cap: Capabilities) -> bool {
+        match self {
+            WorkspaceCommands::FixAll(c) => c.available(cap),
+        }
+    }
+    pub async fn execute(
+        &self,
+        backend: &Backend,
+        args: Vec<serde_json::Value>,
+    ) -> jsonrpc::Result<Option<serde_json::Value>> {
+        match self {
+            WorkspaceCommands::FixAll(c) => {
+                let arg: Result<
+                    <FixAllCommand as WorkspaceCommand>::CommandArgs<'_>,
+                    serde_json::Error,
+                > = serde_json::from_value(serde_json::Value::Array(args));
+                if let Err(e) = arg {
+                    error!("Invalid args passed to {:?}: {e}", c.command_id());
+                    return Err(Error::invalid_request());
+                }
+                let arg = arg.unwrap();
+
+                c.execute(backend, arg).await
+            }
+        }
+    }
+}
+
+pub struct FixAllCommand;
+
+#[derive(Deserialize)]
+pub struct FixAllCommandArg {
+    uri: String,
+}
```

Comment:
As talked about in discord, we could add `Option<FixType>` and fall back to the `SafeFix` or `SafeFixOrSuggestion` in the future (and either do another ServerLinter run or filter current diagnostic)

(note for my future PR, you can close this once you read it)